### PR TITLE
Test cases

### PR DESCRIPTION
**Pull request by JetFuor**
Here are the test cases for the interactors

### CODE REVIEW

File: src/main/java/use_case/chess_puzzle/CheckMoveOutputData.java [hunk 1]
```diff
@@ -5,7 +5,7 @@
  */
 public class CheckMoveOutputData {
     private final String feedback; // For analysis
-    private final boolean isCorrect; // Check correctness
+    public final boolean isCorrect; // Check correctness
```
Comment (KaiYT1) [inline src/main/java/use_case/chess_puzzle/CheckMoveOutputData.java]: these kinds of classes are supposed to have private attributes, with getter and setter methods...

File: src/test/java/CheckMoveInteractorTest.java [hunk 2]
```diff
@@ -0,0 +1,159 @@
...
+import use_case.chess_puzzle.*;
+
+import entity.ChessPuzzle;
+import java.util.List;
+
+import static org.junit.jupiter.api.Assertions.*;
+
+class CheckMoveInteractorTest {
+
+    // ----------- STUB PRESENTER -----------
```
Comment (KaiYT1) [inline src/test/java/CheckMoveInteractorTest.java]: what does "Stub" mean?

File: pom.xml
Comment (KaiYT1) [inline pom.xml]: didn't we already import these dependencies in main?

**Review by KaiYT1 — CHANGES_REQUESTED** [review]

**Review by Eternal128 — APPROVED** [review]
Great work adding test coverage for the ChessPuzzle module. Coverage seems to address both the expected behaviour and several edge cases. Good separation of setup, execution, and assertions! Though be careful in committing unwanted files.
